src/components/AddExperience/index.tsx

```diff
@@ -0,0 +1,273 @@
+import { z } from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useForm } from "react-hook-form";
+import { Button } from "@/components/ui/button";
+import { CalendarIcon } from "lucide-react";
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form";
+import { Input } from "@/components/ui/input";
+import { Checkbox } from "@/components/ui/checkbox";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from "@/components/ui/popover";
+import { cn } from "@/lib/utils";
+import { format } from "date-fns";
+import { Calendar } from "@/components/ui/calendar";
+import { useEffect } from "react";
+import { employmentTypes } from "@/sample-data";
+import type { JobExperienceType } from "@/types";
+
+function AddExperience({
+  onSubmit,
+  onCancel,
+}: {
+  onSubmit: (newExperience: JobExperienceType) => void;
+  onCancel: () => void;
+}) {
+  const formSchema = z
+    .object({
+      title: z
+        .string()
+        .min(2, { message: "Title is a required field" })
+        .max(100),
+      employmentType: z.enum(employmentTypes),
+      company: z
+        .string()
+        .min(2, { message: "Company is a required field" })
+        .max(100),
+      isCurrent: z.boolean(),
+      startDate: z.date({ error: "Start date is a required field" }),
+      endDate: z.date().optional(),
+    })
+    .superRefine((data, ctx) => {
+      if (!data.isCurrent && !data.endDate) {
+        ctx.addIssue({
+          path: ["endDate"],
+          code: "custom",
+          message: "End date is required unless currently employed",
+        });
+      }
+      if (data.endDate && data.startDate > data.endDate) {
+        ctx.addIssue({
+          path: ["endDate"],
+          code: "custom",
+          message: "End date must be after start date",
+        });
+      }
+    });
+
+  const form = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema),
+    defaultValues: {
+      title: "",
+      employmentType: undefined,
+      company: "",
+      isCurrent: false,
+      startDate: undefined,
+      endDate: undefined,
+    },
+  });
+
+  const formValues = form.watch();
```

Comment:
When you call `watch` without any parameter, you're watching all the fields in the form - which triggers a rerender of the component every time a field changes. You should instead watch just for isCurrent, as that's the only one you need - `form.watch('is_current')`